op-chain-ops/cmd/check-l2/main.go

```diff
@@ -276,6 +276,99 @@ func checkPredeployConfig(client *ethclient.Client, name string) error {
 			if err := checkEAS(p, client); err != nil {
 				return fmt.Errorf("EAS: %w", err)
 			}
+
+		case predeploys.Create2DeployerAddr:
+			bytecode, err := bindings.GetDeployedBytecode("Create2Deployer")
+			if err != nil {
+				return err
+			}
+			if err := checkPredeployBytecode(p, "Create2Deployer", client, bytecode); err != nil {
+				return err
+			}
+
+		case predeploys.MultiCall3Addr:
+			bytecode, err := bindings.GetDeployedBytecode("MultiCall3")
+			if err != nil {
+				return err
+			}
+			if err := checkPredeployBytecode(p, "MultiCall3", client, bytecode); err != nil {
+				return err
+			}
+
+		case predeploys.Safe_v130Addr:
+			bytecode, err := bindings.GetDeployedBytecode("Safe_v130")
+			if err != nil {
+				return err
+			}
+			if err := checkPredeployBytecode(p, "Safe_v130", client, bytecode); err != nil {
+				return err
+			}
+
+		case predeploys.SafeL2_v130Addr:
+			bytecode, err := bindings.GetDeployedBytecode("SafeL2_v130")
+			if err != nil {
+				return err
+			}
+			if err := checkPredeployBytecode(p, "SafeL2_v130", client, bytecode); err != nil {
+				return err
+			}
+
+		case predeploys.MultiSendCallOnly_v130Addr:
+			bytecode, err := bindings.GetDeployedBytecode("MultiSendCallOnly_v130")
+			if err != nil {
+				return err
+			}
+			if err := checkPredeployBytecode(p, "MultiSendCallOnly_v130", client, bytecode); err != nil {
+				return err
+			}
+
+		case predeploys.SafeSingletonFactoryAddr:
+			bytecode, err := bindings.GetDeployedBytecode("SafeSingletonFactory")
+			if err != nil {
+				return err
+			}
+			if err := checkPredeployBytecode(p, "SafeSingletonFactory", client, bytecode); err != nil {
+				return err
+			}
+
+		case predeploys.DeterministicDeploymentProxyAddr:
+			bytecode, err := bindings.GetDeployedBytecode("DeterministicDeploymentProxy")
+			if err != nil {
+				return err
+			}
+			if err := checkPredeployBytecode(p, "DeterministicDeploymentProxy", client, bytecode); err != nil {
+				return err
+			}
+
+		case predeploys.MultiSend_v130Addr:
+			bytecode, err := bindings.GetDeployedBytecode("MultiSend_v130")
+			if err != nil {
+				return err
+			}
+			if err := checkPredeployBytecode(p, "MultiSend_v130", client, bytecode); err != nil {
+				return err
+			}
+
+		case predeploys.Permit2Addr:
+			log.Warn("Skipping deployed bytecode check for Permit2 since it contains an immutable Solidity variable")
```

Comment:
Is there a public getter for the chain id? We need an assertion that the chain id is set correctly

Reply:
I believe we will want to compute the domain separator client side and then compare against the value returned in the public getter: https://github.com/Uniswap/permit2/blob/cc56ad0f3439c502c246fc5cfcc3db92bb8b7219/src/EIP712.sol#L26

This will give us assurance that the chainid was set correctly in the bytecode

Reply:
I believe this isn't strictly necessary to have immediately, as manual testing has confirmed everything works as expected and the chain id is being sourced for the L2 config, so it would be hard for a user to get this wrong. 

However, this is a good footgun to test for, so I'd like to merge this PR without addressing this immediately, so that #8650 and #8651 are unblocked. I will address this later this week

Reply:
@tynes are you onboard on unblocking this and get Wyatt to follow-up end of the week with the extra test you asking? Want to make sure we all aligned and things can progress. Thank you so much for helping on this 🙏

Reply:
If we merge this now then we really need to follow up with a test that the value is set correctly or else it will be very bad

Reply:
Ok, I'll implement the test now

Reply:
@tynes I've implemented the test [here](https://github.com/ethereum-optimism/optimism/pull/8285/files#diff-1401051c2ff88b1420ee396fc5464548c3a29fa518c2cb48c7d6b3dd6fb50e05R354-R383)